test/gets.js

```diff
@@ -17,47 +17,46 @@ test('gets', function() {
          '\n' +
-         'The value at position 1 is not a member of ‘TypeRep’.\n');
+         'The value at position 1 is not a member of ‘Function’.\n');
 
-  throws(function() { S.gets(Number, null); },
+  throws(function() { S.gets(S.K(true), null); },
          TypeError,
          'Invalid value\n' +
          '\n' +
-         'gets :: Accessible a => TypeRep -> Array String -> a -> Maybe b\n' +
-         '                                   ^^^^^^^^^^^^\n' +
-         '                                        1\n' +
+         'gets :: Accessible a => Function -> Array String -> a -> Maybe c\n' +
+         '                                    ^^^^^^^^^^^^\n' +
+         '                                         1\n' +
          '\n' +
          '1)  null :: Null\n' +
          '\n' +
          'The value at position 1 is not a member of ‘Array String’.\n');
 
-  throws(function() { S.gets(Number, [], null); },
+  throws(function() { S.gets(S.K(true), [], null); },
          TypeError,
          'Type-class constraint violation\n' +
          '\n' +
-         'gets :: Accessible a => TypeRep -> Array String -> a -> Maybe b\n' +
-         '        ^^^^^^^^^^^^                               ^\n' +
-         '                                                   1\n' +
+         'gets :: Accessible a => Function -> Array String -> a -> Maybe c\n' +
+         '        ^^^^^^^^^^^^                                ^\n' +
+         '                                                    1\n' +
          '\n' +
          '1)  null :: Null\n' +
          '\n' +
          '‘gets’ requires ‘a’ to satisfy the Accessible type-class constraint; the value at position 1 does not.\n');
 
-  eq(S.gets(Number, ['x'], {x: {z: 0}, y: 42}), S.Nothing);
-  eq(S.gets(Number, ['y'], {x: {z: 0}, y: 42}), S.Just(42));
-  eq(S.gets(Number, ['z'], {x: {z: 0}, y: 42}), S.Nothing);
-  eq(S.gets(Number, ['x', 'z'], {x: {z: 0}, y: 42}), S.Just(0));
-  eq(S.gets(Number, ['a', 'b', 'c'], {x: {z: 0}, y: 42}), S.Nothing);
-  eq(S.gets(Number, [], {x: {z: 0}, y: 42}), S.Nothing);
-  eq(S.gets(Object, [], {x: {z: 0}, y: 42}), S.Just({x: {z: 0}, y: 42}));
+  eq(S.gets(S.is(Number), ['x'], {x: {z: 0}, y: 42}), S.Nothing);
+  eq(S.gets(S.is(Number), ['y'], {x: {z: 0}, y: 42}), S.Just(42));
+  eq(S.gets(S.is(Number), ['z'], {x: {z: 0}, y: 42}), S.Nothing);
+  eq(S.gets(S.is(Number), ['x', 'z'], {x: {z: 0}, y: 42}), S.Just(0));
+  eq(S.gets(S.is(Number), ['a', 'b', 'c'], {x: {z: 0}, y: 42}), S.Nothing);
+  eq(S.gets(S.is(Number), [], {x: {z: 0}, y: 42}), S.Nothing);
+  eq(S.gets(S.is(Object), [], {x: {z: 0}, y: 42}), S.Just({x: {z: 0}, y: 42}));
 
-  eq(S.gets(RegExp, ['x'], {x: vm.runInNewContext('/.*/')}), S.Just(/.*/));
-  eq(S.gets(vm.runInNewContext('RegExp'), ['x'], {x: /.*/}), S.Just(/.*/));
+  eq(S.gets(S.K(true), ['x'], {x: vm.runInNewContext('/.*/')}), S.Just(/.*/));
```

Comment:
This is what I had in mind:

```diff
-  eq(S.gets(RegExp, ['x'], {x: vm.runInNewContext('/.*/')}), S.Just(/.*/));
-  eq(S.gets(vm.runInNewContext('RegExp'), ['x'], {x: /.*/}), S.Just(/.*/));
+  eq(S.gets(S.is(RegExp), ['x'], {x: vm.runInNewContext('/.*/')}), S.Just(/.*/));
+  eq(S.gets(S.is(vm.runInNewContext('RegExp')), ['x'], {x: /.*/}), S.Just(/.*/));
```

This way we'd continue to assert that a regexp from one environment is a member of the type represented by `RegExp` from another environment. Admittedly this is really testing `is` rather than `gets`, but that's okay.